### PR TITLE
Fixes #26013 - Add GraphiQL console

### PR DESCRIPTION
**Pull request by xprazak2**
Extracted from  #5336.

It will be much more useful once we have GraphQL :)

### CODE REVIEW

**Comment by timogoebel** [comment]
Lovely, this makes me so happy.
Pro tip: I've been using the [electron based graphql](https://electronjs.org/apps/graphiql) on my mac.

@ekohl: Can you ack? We don't need any packaging for this as it's just available in a dev environment.

**Review by timogoebel — DISMISSED** [review]

File: bundler.d/graphiql.rb [hunk 1]
```diff
@@ -0,0 +1,3 @@
+group :development, :test do
```
Comment (mmoll) [inline bundler.d/graphiql.rb]: why is this needed in `test`?

Comment (timogoebel) [reply bundler.d/graphiql.rb]: I guess it's not needed there. Development should be enough. Good catch.

Comment (xprazak2) [reply bundler.d/graphiql.rb]: I removed it from `:test`

**Review by timogoebel — DISMISSED** [review]

File: bundler.d/graphiql.rb (hunk 1 above)
Comment (mmoll) [reply bundler.d/graphiql.rb]: I guess it should go into `development.rb` then, as it's in that bundler group.

**Comment by lzap** [comment]
Was wondering if we want to create some place with all of these helper apps when in development mode. Some tiny icon because this way only graphql people will use that. Let's advertise this on some page. Maybe we have more of these tools available today, the thing is I don't remember.

image

https://triviahappy.com/articles/welcome-to-the-net-every-webpage-from-the-1995-movie-the-net

**Comment by iNecas** [comment]
@lzap what about submenu in navigation bar?

**Comment by xprazak2** [comment]
@lzap, what do you think?

File: app/registries/menu/loader.rb [hunk 2]
```diff
@@ -94,6 +94,19 @@ def self.load
       Manager.map :labs_menu do |menu|
         menu.sub_menu :lab_features_menu, :caption => N_('Lab Features'), :icon => 'fa fa-flask'
       end
+
+      if Rails.env.development?
+        Manager.map :devel_menu do |menu|
+          menu.sub_menu :devel_tools, :caption => 'Develop tools', :icon => 'pficon pficon-maintenance' do
```
Comment (timogoebel) [inline app/registries/menu/loader.rb]: I would choose a single word as a caption, e.g. `Toolbox`.

Comment (xprazak2) [reply app/registries/menu/loader.rb]: I changed it to `Toolbox`

**Comment by mmoll** [comment]
@xprazak2 tests fail :broken_heart:

**Comment by timogoebel** [comment]
@xprazak2: Can you rebase, please?

**Comment by xprazak2** [comment]
Rebased.

**Review by ekohl — DISMISSED** [review]
:+1: from a packaging perspective.

**Comment by xprazak2** [comment]
I changed a condition when item is added to menu to `Rails.env.development? && defined?(::GraphiQL::Rails::Engine)` because tests on Jenkins bundle without development and then call [rake](https://ci.theforeman.org/job/test_develop_pr_core/21601/database=postgresql,ruby=2.5,slave=fast/console)...

**Review by timogoebel — APPROVED** [review]
Nice! :-)

**Comment by timogoebel** [comment]
@ekohl: Can we please get another ack? Thanks.

**Review by ekohl — DISMISSED** [review]

File: config/routes.rb [hunk 3]
```diff
@@ -549,4 +549,8 @@
       delete 'group/:group' => 'notification_recipients#destroy_group'
     end
   end
+
+  if Rails.env.development?
```
Comment (timogoebel) [inline config/routes.rb]: This also needs `&& defined?(::GraphiQL::Rails::Engine)`

**Review by timogoebel — CHANGES_REQUESTED** [review]

**Review by timogoebel — APPROVED** [review]

**Review by mmoll — APPROVED** [review]

**Comment by mmoll** [comment]
test failures unrelated :green_heart:

**Comment by mmoll** [comment]
merged, díky @xprazak2!

**Comment by lzap** [comment]
Sorry I am late to the party, I love it. Thanks.
